clients/cpp/tests/cli/cli_test_runner.py

```
@@ -189,14 +143,16 @@ def run_cli_smoke_test():
             sys.exit(1)
 
     finally:
+        # Ensure anna processes are cleaned up even if the test fails
         print("Cleaning up...")
-        for proc in procs:
-            try:
-                os.killpg(os.getpgid(proc.pid), signal.SIGTERM)
-            except Exception as e:
-                print(f"Error killing process {proc.pid}: {e}")
+        subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)